File: examples/access/tessellation/point_tessellation.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Tessellate a point location into rectangular tiles aligned with satellite ground tracks.
"""

import brahe as bh

bh.initialize_eop()

# ISS TLE
line1 = "1 25544U 98067A   08264.51782528 -.00002182  00000-0 -11606-4 0  2927"
line2 = "2 25544  51.6416 247.4627 0006703 130.5360 325.0288 15.72125391563537"

# Create propagator and tessellator
prop = bh.SGPPropagator.from_tle(line1, line2, step_size=60.0)
config = bh.OrbitGeometryTessellatorConfig(
    image_width=5000,
    image_length=5000,
    asc_dsc=bh.AscDsc.ASCENDING,
)
tess = bh.OrbitGeometryTessellator(prop, prop.epoch, config, spacecraft_id="ISS")

# Tessellate a point
point = bh.PointLocation(10.0, 30.0, 0.0)
tiles = tess.tessellate_point(point)

print(f"Number of tiles: {len(tiles)}")
for i, tile in enumerate(tiles):
    props = tile.properties
    center = tile.center_geodetic()
    print(f"Tile {i}: center=({center[0]:.4f}, {center[1]:.4f})")
    print(f"  width={props['tile_width']:.0f} m, length={props['tile_length']:.0f} m")

File: examples/access/tessellation/point_tessellation.rs
//! Tessellate a point location into rectangular tiles aligned with satellite ground tracks.
//! Demonstrates point tessellation with an ascending-only configuration.

#[allow(unused_imports)]
use brahe as bh;
use bh::access::constraints::AscDsc;
use bh::access::location::{AccessibleLocation, PointLocation};
use bh::access::tessellation::{OrbitGeometryTessellator, OrbitGeometryTessellatorConfig, Tessellator};
use bh::propagators::SGPPropagator;

fn main() {
    bh::initialize_eop().unwrap();

    // ISS TLE
    let line1 = "1 25544U 98067A   08264.51782528 -.00002182  00000-0 -11606-4 0  2927";
    let line2 = "2 25544  51.6416 247.4627 0006703 130.5360 325.0288 15.72125391563537";

    // Create propagator and tessellator
    let prop = SGPPropagator::from_tle(line1, line2, 60.0).unwrap();
    let epoch = prop.epoch;
    let config = OrbitGeometryTessellatorConfig::default()
        .with_asc_dsc(AscDsc::Ascending);
    let tess = OrbitGeometryTessellator::new(
        Box::new(prop), epoch, config, Some("ISS".to_string()),
    );

    // Tessellate a point
    let point = PointLocation::new(10.0, 30.0, 0.0);
    let tiles = tess.tessellate(&point).unwrap();

    println!("Number of tiles: {}", tiles.len());
    for (i, tile) in tiles.iter().enumerate() {
        let center = tile.center_geodetic();
        let props = tile.properties();
        println!("Tile {}: center=({:.4}, {:.4})", i, center[0], center[1]);
        println!("  width={:.0} m, length={:.0} m",
            props["tile_width"].as_f64().unwrap(),
            props["tile_length"].as_f64().unwrap(),
        );
    }

}

